File: backend/src/main/java/com/retypeme/project/chain/Runner.java
package com.retypeme.project.chain;

import static org.web3j.tx.Contract.GAS_LIMIT;
import static org.web3j.tx.ManagedTransaction.GAS_PRICE;

import com.retypeme.project.chain.seed.ChainSeed;
import com.retypeme.project.chain.contract.HelloWorld;
import com.retypeme.project.chain.seed.PoligonChainSeed;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;
import org.web3j.crypto.Credentials;
import org.web3j.protocol.Web3j;
import org.web3j.tx.RawTransactionManager;
import org.web3j.tx.TransactionManager;
import org.web3j.tx.gas.StaticGasProvider;

@Component
public class Runner implements CommandLineRunner {

    static Logger logger = LoggerFactory.getLogger(CommandLineRunner.class);

    private HelloWorld deploy(ChainSeed seed, String privateKey) throws Exception {
        Credentials credentials = Credentials.create(privateKey);
        Web3j web3j = Web3j.build(seed.getHttpServer());
        TransactionManager tm =
            new RawTransactionManager(web3j, credentials, seed.getChainId());
        StaticGasProvider staticGasProvider = new StaticGasProvider(GAS_PRICE, GAS_LIMIT);
        return HelloWorld.deploy(web3j, tm, staticGasProvider).send();
    }

    @Value("${contract.chain.id}")
    private int chainId;
    @Value("${contract.private.key}")
    private String privateKey;
    @Value("${contract.network.url}")
    private String networkUrl;


    @Override
    public void run(String... args) throws Exception {
        ChainSeed seed = new PoligonChainSeed(networkUrl, chainId);
        HelloWorld contract = deploy(seed, privateKey);

        String contractAddress = contract.getContractAddress();
        logger.info("contract address: " + contractAddress);
    }
}

File: backend/src/main/kotlin/com/retypeme/backend/Runner.kt
package com.retypeme.backend

import com.retypeme.backend.contract.Acc
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component
import org.web3j.crypto.Credentials
import org.web3j.protocol.Web3j
import org.web3j.protocol.http.HttpService
import org.web3j.tx.Contract.GAS_LIMIT
import org.web3j.tx.Contract.GAS_PRICE
import java.math.BigInteger

class Runner : CommandLineRunner {

    @Value("\${contract.address}")
    private val contractAddress: String? = null

    @Value("\${contract.private-key}")
    private val privateKey: String? = null

    private val logger = LoggerFactory.getLogger(Runner::class.java)

    override fun run(vararg args: String?) {
        val web3 = Web3j.build(HttpService())
        val web3ClientVersion = web3.web3ClientVersion().send()
        val clientVersion = web3ClientVersion.web3ClientVersion
        logger.info("Client version: $clientVersion")

        val contract = Acc.load(contractAddress, web3, Credentials.create(privateKey), GAS_PRICE, GAS_LIMIT)

        var result = contract.myValue().send()
        logger.info("Result of myValue(): $result")

        val newValue = BigInteger.valueOf(42)
        contract.addValue(newValue).send()

        result = contract.myValue().send()
        logger.info("Updated result of getSomeValue(): $result")
    }
}